src/nbsphinx/__init__.py

```diff
@@ -1689,17 +1689,37 @@ def html_page_context(app, pagename, templatename, context, doctree):
         app.add_css_file('nbsphinx-gallery.css')
 
 
+def backwards_compat_overwrite(copyfile=sphinx.util.copyfile):
+    """Return kwargs dictionary to pass to copyfile() for consistent behavior
+
+    Before version 8 of Sphinx, the default behavior of the copyfile function
+    was to overwrite the file at the target path if it already existed.
+    Version 8 requires passing force=True.
+
+    Ref: https://github.com/sphinx-doc/sphinx/pull/12647
+    """
+    from inspect import signature
+    if "force" in signature(copyfile).parameters:
+        return {"force": True}
+    else:
+        return {}
+
+
 def html_collect_pages(app):
     """This event handler is abused to copy local files around."""
     files = set()
     for file_list in app.env.nbsphinx_files.values():
         files.update(file_list)
+    overwrite = backwards_compat_overwrite()
     for file in status_iterator(files, 'copying linked files... ',
                                 sphinx.util.console.brown, len(files)):
         target = os.path.join(app.builder.outdir, file)
         sphinx.util.ensuredir(os.path.dirname(target))
         try:
-            sphinx.util.copyfile(os.path.join(app.env.srcdir, file), target)
+            sphinx.util.copyfile(
+                os.path.join(app.env.srcdir, file),
+                target,
+                **overwrite)
```

Comment:
Now that we have this as a function, you could get rid of the additional line above by directly calling the function here (and again below):

```suggestion
                **backwards_compat_overwrite())
```

What do you think about this?